### DISCUSSION TITLE
how to refresh token in vue app

### DISCUSSION THREAD

**Discussion by BonderTam** [logto-io/logto#1950] [Q&A]
I prefer keeping token expiration time in 60s within Logto,  
after
`
const { getAccessToken } = useLogto();
const token = await getAccessToken();
` 
how to auto-refresh token while user keep online.

**Comment by charIeszhao** [comment]
Hi @BonderTam , thanks for reaching out.

If I understand it correctly, you would like to set accessToken lifetime to 60s, and auto-refresh it whenever you're making a new request and the accessToken is expired, am I right?

If that's the case then it's automatically handled by the Vue SDK, as long as your refreshToken is still valid. You only need to call `getAccessToken()`and the SDK will issue a new accessToken if the previous one is expired.

Let me know if you have further questions. And you can also join our [discord channel](https://discord.gg/yRZhfP8p) for instant release notification and direct dev team support.

Thanks,
Charles

**Comment by BonderTam** [comment]
yes, call getAccessToken() is only thing that i can do, within vue's setup function.
i hadn't tried to find out very carefully why  got 'invalid_token'  error since the account  had signed in.
but is that possiable to provide a without setup function version , something like:  
`const { getAccessTokenWithout } = useLogtoWithout(); `
so we can get token everywhere,  and first of all  we should had a store,  something like : pinia.
Is this correct? 

For i am doing now , is setting a setInterval function in an empty vue component,  to get 3 accessTokens of different API Resouce.
I feel stupid.

**Comment by charIeszhao** [comment]
Well, as far as I know, we can't provide a function to be called outside of Vue 3's "setup script".

I would like to know more details about your "Invalid token" issue. Can you please provide more details on that, maybe the entire error message? Or a minimum reproducible project would be great enough to let me quickly jump into your issues.

As for the access token store, yes, we already have built-in support in our JS SDKs, you only need to pass `resource` to `getAccessToken` function to get a resource specific access token for you.

<img width="414" alt="image" src="https://user-images.githubusercontent.com/12833674/190850352-ed8d480f-db8c-4d16-a80b-24f2365513f3.png">

Feel free to ask me if you have any further questions. And since we have also connected on discord, you can ping me directly, too.

Thanks,
Charles
